File: app/src/main/java/com/prideven/android/HungryEats/menuItems/CustomAdapter.java
package com.prideven.android.hungryeats.menuitems;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.recyclerview.widget.RecyclerView;

import com.prideven.android.hungryeats.GetCartValuesListener;
import com.prideven.android.hungryeats.MenuItem;
import com.prideven.android.hungryeats.R;

import java.util.ArrayList;
import java.util.List;

public class CustomAdapter extends RecyclerView.Adapter<CustomViewHolder> {
    private static final String TAG = "CustomAdapter";

    List<MenuItem> items = new ArrayList<>();
    GetCartValuesListener getCartValuesListener;

    public CustomAdapter(List<MenuItem> dataSet,GetCartValuesListener getCartValuesListener) {
        this.items = dataSet;
        this.getCartValuesListener=getCartValuesListener;
    }

    @Override
    public CustomViewHolder onCreateViewHolder(ViewGroup viewGroup, int viewType) {
        // Create a new view.
        View v = LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.restaurant_menu_item_layout, viewGroup, false);

        return new CustomViewHolder(v);
    }

    @Override
    public void onBindViewHolder(CustomViewHolder viewHolder, final int position) {
        Log.d(TAG, "Element " + position + " set.");


        viewHolder.setData(items.get(position),getCartValuesListener);
    }

    @Override
    public int getItemCount() {
        if(items == null)
            return 0;
        return items.size();
    }
}

File: app/src/main/java/com/prideven/android/HungryEats/menuItems/CustomAdapter.kt
package com.prideven.android.hungryeats.menuitems

import android.util.Log
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.prideven.android.hungryeats.GetCartValuesListener
import com.prideven.android.hungryeats.MenuItem
import com.prideven.android.hungryeats.R
import java.util.*

class CustomAdapter(
    dataSet: List<MenuItem>?,
    getCartValuesListener: GetCartValuesListener
) : RecyclerView.Adapter<CustomViewHolder>() {
    var items: List<MenuItem>? =
        ArrayList()
    var getCartValuesListener: GetCartValuesListener
    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): CustomViewHolder {
        // Create a new view.
        val v = LayoutInflater.from(viewGroup.context)
            .inflate(R.layout.restaurant_menu_item_layout, viewGroup, false)
        return CustomViewHolder(v)
    }

    override fun onBindViewHolder(viewHolder: CustomViewHolder, position: Int) {
        Log.d(TAG, "Element $position set.")
        viewHolder.setData(items!![position], getCartValuesListener)
    }

    override fun getItemCount(): Int {
        return if (items == null) 0 else items!!.size
    }

    companion object {
        private const val TAG = "CustomAdapter"
    }

    init {
        items = dataSet
        this.getCartValuesListener = getCartValuesListener
    }
}